scriptHelper.js

```diff
@@ -1,43 +1,176 @@
 // Write your helper functions here!
-require('isomorphic-fetch');
-
-function addDestinationInfo(document, name, diameter, star, distance, moons, imageUrl) {
-   // Here is the HTML formatting for our mission target div.
-   /*
-                <h2>Mission Destination</h2>
-                <ol>
-                    <li>Name: </li>
-                    <li>Diameter: </li>
-                    <li>Star: ${star}</li>
-                    <li>Distance from Earth: </li>
-                    <li>Number of Moons: </li>
-                </ol>
-                <img src="">
-   */
+
+require("isomorphic-fetch");
+
+function addDestinationInfo(
+  document,
+  name,
+  diameter,
+  star,
+  distance,
+  moons,
+  imageUrl
+) {
+  // Here is the HTML formatting for our mission target div.
+  let missionTarget = document.querySelector("#missionTarget");
+  missionTarget.innerHTML = `<h2>Mission Destination</h2>
+          <ol>
+            <li>Name: ${name}</li>
+            <li>Diameter: ${diameter}</li>
+            <li>Star: ${star}</li>
+            <li>Distance from Earth: ${distance} </li>
+           <li>Number of Moons: ${moons}</li>
+          </ol>
+          <img src="${imageUrl}">`;
 }
+//addDestinationInfo(pickPlanets(planets));
 
 function validateInput(testInput) {
-   
+  //validateInput() should take in a string as a parameter and return "Empty", "Not a Number", or "Is a Number" as appropriate.
+  if (testInput === "") {
+    return "Empty";
+  } else if (!isNaN(testInput)) {
+    return "Is a Number";
+  } else if (typeof String(testInput) === "string") {
+    return "Not a Number";
+  }
 }
 
+//you will use validateInput() to complete the formSubmission() function.
 function formSubmission(document, list, pilot, copilot, fuelLevel, cargoLevel) {
-   
+  //console.log("formSubmission");
+  //if any of these input variables is blank, throw and alert
+  if (
+    validateInput(pilot) === "Empty" ||
+    validateInput(copilot) === "Empty" ||
+    validateInput(fuelLevel) === "Empty" ||
+    validateInput(cargoLevel) === "Empty"
+  ) {
+    alert("All fields are required!");
+  }
+  //Pilot and CoPilot Names should be a String, and Fuel Level and Cargo Mass should be a Number
+  if (
```

Comment:
```suggestion
  else if (
```
When testing your page in the browser, I noticed that if I don't enter anything in the inputs and submit the form, both the empty field and enter valid data alerts show up. I'd recommend using an else if instead of an if on lines 52 and 63 to ensure that the most relevant alert is rendered vs. multiple ones. Also, else-if ensures that the `faultyItems` list does not become visible until you can the empty field and data type validations are done.